devito/archinfo.py

```diff
@@ -60,18 +62,36 @@ def get_cpu_brand():
             except:
                 pass
 
-            return None
+            try:
+                return cpuinfo.get_cpu_info().get('raw_arch_string')
+            except:
+                pass
 
         cpu_info['flags'] = get_cpu_flags()
         cpu_info['brand'] = get_cpu_brand()
 
-    if not all(i in cpu_info for i in ('flags', 'brand')):
-        # Fallback
-        ci = cpuinfo.get_cpu_info()
-        cpu_info['flags'] = ci.get('flags')
-        cpu_info['brand'] = ci.get('brand')
+    try:
+        if 'flags' not in cpu_info:
+            # Fallback
+            ci = cpuinfo.get_cpu_info()
+            cpu_info['flags'] = ci.get('flags')
+    except:
+        pass
 
     # Detect number of logical cores
+    try:
```

Comment:
why do you need a try except here? I think you don't

Reply:
Ah... you need it in case `lscpu` fails?

Reply:
yes

Reply:
well... I guess it cannot fail though...so I will remove...

Reply:
it  can -- if there's no `lscpu` on the system

Reply:
But we are not aware of such a case so far right?